src/main/cpp/blaze.cc

```diff
@@ -1426,10 +1426,24 @@ static map<string, EnvVarValue> PrepareEnvironmentForJvm() {
 
     // Disable HTTP proxies for localhost and any localhost-like address,
     // in case we (or gRPC, etc.) ever use one of these addresses.
-    std::string localhost_addresses = "localhost,127.0.0.1,0:0:0:0:0:0:0:1,::1";
+    string localhost_addresses = blaze::GetEnv("no_proxy");
+    vector<string> localhost_addresses_vec{"localhost", "127.0.0.1", "0:0:0:0:0:0:0:1", "::1"};
+    if (localhost_addresses.empty()) {
+      localhost_addresses = blaze::GetEnv("NO_PROXY");
+    }
+    for (auto const& s : localhost_addresses_vec) {
+      if (localhost_addresses.find(s) == string::npos) {
+        localhost_addresses += ",";
+        localhost_addresses += s;
+      }
+    }
+
     result["no_proxy"] = EnvVarValue(EnvVarAction::SET, localhost_addresses);
     result["NO_PROXY"] = EnvVarValue(EnvVarAction::SET, localhost_addresses);
 
+    BAZEL_LOG(WARNING)
```

Comment:
I think this should be INFO rather than WARNING. This is informative and not useful to know for any individual build.